src/CachePlugin.php

```diff
@@ -20,29 +22,26 @@ class CachePlugin implements Plugin
     private $pool;
 
     /**
-     * Default time to store object in cache. This value is used if CachePlugin::respectCacheHeaders is false or
-     * if cache headers are missing.
-     *
-     * @var int
+     * @var StreamFactory
      */
-    private $defaultTtl;
+    private $streamFactory;
 
-    /**
-     * Look at the cache headers to know how long this response is going to be cached.
-     *
-     * @var bool
-     */
-    private $respectCacheHeaders;
+    private $config = [
+        'default_ttl' => null,
+        'respect_cache_headers' => true,
+    ];
 
     /**
      * @param CacheItemPoolInterface $pool
-     * @param array                  $options
+     * @param StreamFactory          $streamFactory
+     * @param array                  $config
      */
-    public function __construct(CacheItemPoolInterface $pool, array $options = [])
+    public function __construct(CacheItemPoolInterface $pool, StreamFactory $streamFactory, array $config = [])
```

Comment:
None, but SocketHttpClient was using `$config` so I thought I would be consistent.